### PR TITLE
Include the `expect_identical_linter`

### PR DESCRIPTION
**Pull request by RasmusSkytte**
### Intent
This PR introduces the `expect_identical_linter` to our set of linters.

This should help make our tests even more specific.

### Approach
Exception removed for the `expect_identical_linter`.

Existing `expect_equal` converted to `expect_identical` unless we are comparing functions.
Functions also carry their own environment, which makes them equal but not "identical",
so here we keep using `expect_equal`.

### Known issues
N/A

### Checklist

* [x] The PR passes all local unit tests
* [ ] I have documented any new features introduced
* [ ] If the PR adds a new feature, please add an entry in `NEWS.md`
* [ ] A reviewer is assigned to this PR

### CODE REVIEW

File: tests/testthat/test-DiseasySeason.R [hunk 1]
```diff
@@ -196,13 +196,12 @@ test_that("$use_covid_season_v2() works", {
                            observables = observables)
 
     # Default scale
-    models <- s$get_covid_season_v1()
```
Comment (kaare-gr) [inline tests/testthat/test-DiseasySeason.R:199]: So now only v2 is tested...

**Review by kaare-gr — APPROVED** [review]
Should tests automatically probe for possible versions and test all of these automatically? e.g. in seasons.

File: tests/testthat/test-DiseasySeason.R (hunk 1 above)
Comment (RasmusSkytte) [reply tests/testthat/test-DiseasySeason.R:199]: It used to be that v1 was tested. I fixed the type in the test to now (correctly) test v2 instead
"models <- s$get_covid_season_v1()" -> "models <- s$get_covid_season_v2()"

**Comment by RasmusSkytte** [comment]
> Should tests automatically probe for possible versions and test all of these automatically? e.g. in seasons.

We could think about more dynamically testing, but that would fall outside the scope of this PR.

With the code-coverage workflows, we do test how much of the code base we test, so we will know if we forget to test a season model. However, we could do it dynamically as you suggest. I can make a issue to track it.

**Pull request merged**